libdnf/module/module_sack.cpp

```diff
@@ -265,6 +266,9 @@ ModuleSack::Impl::collect_data_for_modular_filtering() {
 void ModuleSack::Impl::module_filtering() {
     auto [include_NEVRAs, exclude_NEVRAs, names, src_names, reldep_name_list] = collect_data_for_modular_filtering();
 
+    auto platform_id = detect_platform_name_and_stream();
+    ModuleItem::create_platform_solvable(module_sack->get_weak_ptr(), platform_id.first, platform_id.second);
```

Comment:
I think you are correct. This is the best place to detect platform ID. Because the detection requires all RPM metadata loaded, therefore we cannot call it earlier. I only want to ask you to ensure that the detection will be performed only once. In future we will call modular filtering method multiple times therefore we have to avoid adding platform more than once.